env/AWSPW.env

```diff
@@ -19,96 +19,64 @@ export NTHSTACK=1024000000
 ulimit -s unlimited
 ulimit -a
 
-if [[ "${step}" = "fcst" ]] || [[ "${step}" = "efcs" ]]; then
+# Calculate common variables
+# Check first if the dependent variables are set
+if [[ -n "${ntasks:-}" && -n "${max_tasks_per_node:-}" && -n "${tasks_per_node:-}" ]]; then
+    max_threads_per_task=$((max_tasks_per_node / tasks_per_node))
+    NTHREADSmax=${threads_per_task:-${max_threads_per_task}}
+    NTHREADS1=${threads_per_task:-1}
+    [[ ${NTHREADSmax} -gt ${max_threads_per_task} ]] && NTHREADSmax=${max_threads_per_task}
+    [[ ${NTHREADS1} -gt ${max_threads_per_task} ]] && NTHREADS1=${max_threads_per_task}
+    APRUN="${launcher} -n ${ntasks}"
+fi
```

Comment:
There are a handful of jobs that do not set these variables because they don't source config.resources.  In particular, `gfsgempakncdccupapgif` and `gfsnpoess_pgrb2_0p5deg`.  If the `WAFS` tasks were ever added to the mesh, they would also run into this issue.  Since these only run on WCOSS2, I could add an `elif` block in `WCOSS2.env` to check if one of these jobs is sourcing it or otherwise raise an error.  For the other `env` files, I will just raise an error.

Reply:
WAFS is definitely (may be) not going to be added back.

Reply:
Those jobs' configs should have sourcing of `config.resources` added like all the other jobs, with blocks added to the env files as well if needed.